ddtrace/utils/formats.py

```diff
@@ -97,7 +101,22 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    tags = []
+    if tags_str.count(":") == 1 and "," not in tags_str:
```

Comment:
If we are worried about spaces appearing around separators, would it maybe make sense to try and normalise the string first? That is, remove excess white spaces around `:`s and trim on both ends?

Reply:
Problem is that we must allow spaces in tag names and values

Reply:
May I suggest something along the lines of

~~~ python
test = "key: val key2:val2 bubbles:fizzy"

fc = "".join([c for c in test if c in {":", ","}])
fs = " ".join("".join([c for c in test if c in {":", " "}]).split())

n = len([c for c in test if c == ":"])

if n == len(fc.split(",")):
    print("comma separated")
elif n == len(fs.split(" ")):
    print("space separated")
else:
    print("error :(")
~~~